packages/dev/core/src/Culling/ray.ts

```diff
@@ -798,7 +801,8 @@ Scene.prototype._internalPick = function (
             continue;
         }
 
-        const world = mesh.getWorldMatrix();
+        const forceCompute = multiCamera && mesh.isWorldMatrixCameraDependent();
+        const world = mesh.computeWorldMatrix(forceCompute, currentCamera);

```

Comment:
Note that if we don't want a breaking change, we should recompute the world matrix with the scene active camera for meshes with `forceCompute=true` after we have performed the picking...